test/specs/rules/disallow-spaces-in-function-declaration.js

```diff
@@ -40,4 +40,30 @@ describe('rules/disallow-spaces-in-function-declaration', function() {
             assert(checker.checkString('function abc (){}').isEmpty());
         });
     });
+
+    describe('es6', function() {
+        beforeEach(function() {
+            checker.configure({ esnext: true });
+        });
+
+        it('should not report missing space before round brace in export default function', function() {
+            checker.configure({ disallowSpacesInFunctionDeclaration: { beforeOpeningRoundBrace: true } });
+            assert(checker.checkString('export default function() {}').isEmpty());
+        });
+
+        it.skip('should report missing space before round brace in export default function', function() {
+            checker.configure({ disallowSpacesInFunctionDeclaration: { beforeOpeningRoundBrace: true } });
+            assert(checker.checkString('export default function(){}').getErrorCount() === 1);
+        });
+
+        it('should not report missing space before curly brace in export default function', function() {
```

Comment:
Same comment as above: move to its own describe and use a beforeEach for what's common